lanelet2_python/python_api/core.cpp

```diff
@@ -858,150 +1196,264 @@ BOOST_PYTHON_MODULE(PYTHON_API_MODULE_NAME) {  // NOLINT
+           "a unique value.")
+      .def("add", selectAdd<LineString3d>(), arg("linestring"),
+           "Add a linestring and all its points to the map. It its (or its points IDs) are zero, it will be set to a "
+           "unique value.")
+      .def("add", selectAdd<Polygon3d>(), arg("polygon"),
+           "Add a polygon and all its points to the map. It its (or its points IDs) are zero, it will be set to a "
+           "unique value.")
+      .def("add", selectAdd<const RegulatoryElementPtr&>(), arg("regelem"),
+           "Add a regulatory element and all its subobjects to the map. It its (or its subobjects IDs) are zero, it "
+           "will be set to a unique value.");
   register_ptr_to_python<LaneletMapConstPtr>();
 
   class_<LaneletSubmap, bases<LaneletMapLayers>, LaneletSubmapPtr, boost::noncopyable>(
-      "LaneletSubmap", "Object for managing parts of a lanelet map", init<>("LaneletSubmap()"))
-      .def(
-          "laneletMap", +[](LaneletSubmap& self) { return LaneletMapPtr{self.laneletMap()}; })
-      .def("add", selectSubmapAdd<Point3d>())
-      .def("add", selectSubmapAdd<Lanelet>())
-      .def("add", selectSubmapAdd<Area>())
-      .def("add", selectSubmapAdd<LineString3d>())
-      .def("add", selectSubmapAdd<Polygon3d>())
-      .def("add", selectSubmapAdd<const RegulatoryElementPtr&>());
+      "LaneletSubmap",
+      "A submap manages parts of a lanelet map. An important difference is that adding an object to the map will "
+      "*not* "
+      "add its subobjects too, making it more efficient to create. Apart from that, it offers a similar "
+      "functionality "
+      "compared to a lanelet map.",
+      init<>("Create an empty lanelet submap"))
+      .def(
+          "laneletMap", +[](LaneletSubmap& self) { return LaneletMapPtr{self.laneletMap()}; },
+          "Create a lanelet map of this submap that also holds all subobjects. This is a potentially costly "
+          "operation.")
+      .def("add", selectSubmapAdd<Point3d>(), arg("point"), "Add a point to the submap")
+      .def("add", selectSubmapAdd<Lanelet>(), arg("lanelet"),
+           "Add a lanelet to the submap, without its subobjects. If its ID is zero it will be set to a unique value "
+           "instead.")
+      .def("add", selectSubmapAdd<Area>(), arg("area"),
+           "Add an area to the submap, without its subobjects. If its ID is zero it will be set to a unique value "
+           "instead.")
+      .def("add", selectSubmapAdd<LineString3d>(), arg("linestring"),
+           "Add a linesting to the submap, without its points. If its ID is zero it will be set to a unique value "
+           "instead.")
+      .def("add", selectSubmapAdd<Polygon3d>(), arg("polygon"),
+           "Add a polygon to the submap, without its points. If its ID is zero it will be set to a unique value "
+           "instead.")
+      .def("add", selectSubmapAdd<const RegulatoryElementPtr&>(), arg("regelem"),
```

Comment:
```suggestion
      .def("add", selectSubmapAdd<const RegulatoryElementPtr&>(), arg("regElem"),
```

Reply:
Hm, currently we use `regElem` and `regelem` equally, I'd say. Not sure what I like more. One the one hand we also use `linestring` instead of `lineString`, on the other it's still an abbreviation, while linestring isn't.
I'll change it to `regelem` everywhere, but you can also convince me of the opposite.

Reply:
I’d argue „linestring“ is somewhat a word, while „regElem“ makes more obvious that it’s our abbrev. for regulatory element. But no strong feelings 😅

Reply:
valid point, but not convincing enough to cause me to redo it :wink: